src/Java.Interop/Java.Interop/JniRuntime.JniValueManager.cs

```diff
@@ -527,6 +527,14 @@ public JniValueMarshaler GetValueMarshaler (Type type)
 				if (typeof (IJavaPeerable).GetTypeInfo ().IsAssignableFrom (info)) {
 					return JavaPeerableValueMarshaler.Instance;
 				}
+
+				foreach (var iface in info.ImplementedInterfaces) {
+					var ifaceInfo = iface.GetTypeInfo ();
+					marshalerAttr   = ifaceInfo.GetCustomAttribute<JniValueMarshalerAttribute> ();
```

Comment:
Does this actually make sense?  I'm not sure it does.

This states that an interface can contain a `[JniValueMarshaler]` custom attribute on interfaces -- which is probably a mistake, as we'll see! -- but what *should* happen if a type implements more than one interface, each of which has a custom marshaler?

```csharp
[JniValueMarshaler (...)]
public interface IX {}

[JniValueMarshaler (...)]
public interface IY {}

public class Example : JavaObject, IX, IY {
  // ...
}
```

With this PR, we'll wind up using the custom marshaler from the first interface.  Is this correct?  Is this even *sane*?

I think we might instead want to *prevent* allowing interfaces to have custom marshalers, and just restrict to classes/structs/etc.